include/ghex/communication_object_2.hpp

```diff
@@ -22,6 +22,10 @@
 #include <stdio.h>
 #include <functional>
 
+// define this macro for fat callbacks
+// may not yet work optimally with the current ucx implementation
+//#define GHEX_COMM_OBJ_USE_FAT_CALLBACKS
```

Comment:
Where are these FAT call backs described? what are they?

Reply:
I will extend this explanation

Reply:
ok, now I understood. Good point, have changed it accordingly.